tests/e2e/controllers/messages.js

```diff
@@ -0,0 +1,208 @@
+  url.pathname = endpoint;
+  url = Url.format(url);
+  return request({ uri: url, json: true, qs: queryParams });
+};
+
+describe('messages controller', function() {
+  var data_record_rev;
+  beforeEach(function(done) {
+    utils.beforeEach()
+      .then(() => adminDb.put(data_record))
+      .then((result) => {
+        data_record_rev = result.rev;
+        done();
+      })
+      .catch(done);
+  });
+
+  afterEach(function(done) {
+    adminDb.remove(data_record._id, data_record_rev)
+      .then(() => done())
+      .catch(done);
+  });
+
+  it('should fetch all messages', function(done) {
+    apiRequest('/api/v1/messages')
+    .then((result) => {
+      // TODO stop emitting everything twice : https://github.com/medic/medic-webapp/issues/3400
+      // assert.equal(result.length, 3);
+      assert.equal(result.length, 6);
+      done();
+    })
+    .catch((err) => done(err));
+  });
+
+  it('should fetch messages in ascending order', function(done) {
+    apiRequest('/api/v1/messages')
+    .then((result) => {
+      assertSortedByAscendingDate(result);
+      done();
+    })
+    .catch((err) => done(err));
+  });
+
+  it('should fetch messages in descending order', function(done) {
+    apiRequest('/api/v1/messages', { descending: true })
+    .then((result) => {
+      assertSortedByDescendingDate(result);
+      done();
+    })
+    .catch((err) => done(err));
+  });
+
+  it('should fetch messages by state', function(done) {
+    apiRequest('/api/v1/messages', { state: 'pending' })
+    .then((result) => {
+      assert.equal(result.length, 1);
+      assert.equal(result[0].id, pendingId);
+      done();
+    })
+    .catch((err) => done(err));
+  });
+
+  it('should fetch messages by multiple states', function(done) {
+    apiRequest('/api/v1/messages', { states: [ 'pending', 'sent' ] })
+    .then((result) => {
+      assert.equal(result.length, 3); // fetches all
+      done();
+    })
+    .catch((err) => done(err));
+  });
+
+  it('should fetch messages by state in descending order', function(done) {
+    apiRequest('/api/v1/messages', { state: 'pending', descending: true })
+    .then((result) => {
+      assertSortedByDescendingDate(result);
+      done();
+    })
+    .catch((err) => done(err));
+  });
+
+  it('should fetch messages by multiple states in descending order', function(done) {
+    apiRequest('/api/v1/messages', { states: [ 'pending', 'sent' ], descending: true })
+    .then((result) => {
+      assertSortedByDescendingDate(result);
+      done();
+    })
+    .catch((err) => done(err));
+  });
+
+  var assertSortedByDescendingDate = (arr) => {
+    for (let i = 1; i < arr.length; i++) {
+      assert.ok(
+        arr[i-1].sending_due_date >=
+        arr[i].sending_due_date,
+        'Expected ' + arr[i-1].sending_due_date + ' >= ' + arr[i].sending_due_date);
+    }
+  };
+
+  var assertSortedByAscendingDate = (arr) => {
+    for (let i = 1; i < arr.length; i++) {
+      assert.ok(
```

Comment:
You can use [assert.isBelow](http://chaijs.com/api/assert/#method_isbelow) here which might print you a pretty error message for free!